src/Shared/UnitTests/FileUtilities_Tests.cs

```diff
@@ -243,7 +244,10 @@ public void HasExtension_WhenFileNameHasExtension_ReturnsTrue(string fileName, s
         {
             var result = FileUtilities.HasExtension(fileName, allowedExtensions);
 
-            Assert.True(result);
+            if (!FileUtilities.GetIsFileSystemCaseSensitive() || allowedExtensions.All(extension => fileName.Contains(extension)))
```

Comment:
allowedExtensions.Any? Otherwise, it'll never be true if there are multiple distinct extensions as in the last case.